### PR TITLE
Feature/integration tests/change settings

### PR DESCRIPTION
**Pull request by genti94**
Finished changeSettings file

### CODE REVIEW

**Comment by FinalAngel** [comment]
same here @genti94 pls land it :)

**Comment by vxsx** [comment]
@FinalAngel this one depends on https://github.com/FinalAngel/django-cms/pull/130 and is not safe to land yet.

but today

File: cms/tests/frontend/integration/changeSettings.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    this.click('#id_language');
+
+                    //selects German language
+                    this.fill ('#usersettings_form', {
+                        'language': 'de'
+                    }, true);
+                })
+        })
+
+        .waitForResource (/admin/)
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: unnecessary space

File: cms/tests/frontend/integration/changeSettings.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                //checks h1 tag if the language changed
+                .waitForSelector('#content-main', function () {
+                    test.assertExists('#content h1', 'h1 tag with german title exists');
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: it exists in any case, it doesn't mean that the language was changed

File: cms/tests/frontend/integration/changeSettings.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                    this.echo(this.fetchText('#content h1'));
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: this is not necessary

File: cms/tests/frontend/integration/changeSettings.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        .waitForResource (/admin/)
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                 //checks h1 tag if the language changed
+                .waitForSelector('#content-main', function () {
+                    test.assertExists('#content h1', 'h1 tag with english title exists');
+                    this.echo(this.fetchText('#content h1'));
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: here as well

File: cms/tests/frontend/integration/changeSettings.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        })
+        // waits till Sideframe is open
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                .waitForSelector('#content-main', function () {
+
+                    test.assertExists('#id_language', 'language button exists');
+
+                    this.click('#id_language');
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: why click on it btw?

File: cms/tests/frontend/integration/changeSettings.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                //checks h1 tag if the language changed
+                .waitForSelector('#content-main', function () {
+                    test.assertEquals(this.fetchText('#content h1'), 'Benutzer-Einstellung ändern', 'German language applied');
+                })
+
+                .waitForSelector('#content-main', function () {
+
+                    test.assertExists('#id_language', 'Menu correct reloaded');
+
+                    this.click('#id_language');
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: and here

File: cms/tests/frontend/integration/changeSettings.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+        .waitForResource(/admin/)
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                //checks h1 tag if the language changed
+                .waitForSelector('#content-main', function () {
+                    test.assertEquals(this.fetchText('#content h1'), 'Benutzer-Einstellung ändern', 'German language applied');
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: line is too long

File: cms/tests/frontend/integration/changeSettings.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+        .waitForResource(/admin/)
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                 //checks h1 tag if the language changed
+                .waitForSelector('#content-main', function () {
+                    test.assertEquals(this.fetchText('#content h1'), 'Change user setting', 'English language applied');
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: line is too long

File: cms/tests/frontend/integration/changeSettings.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        // click on User Settings
+        .waitUntilVisible('.cms-toolbar-item-navigation-hover a', function () {
+            this.click('.cms-toolbar-item-navigation-hover a[href$="/admin/cms/usersettings/"]');
+        })
+        // waits till Sideframe is open
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                .waitForSelector('#content-main', function () {
+
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/changeSettings.js]: remove space

File: cms/tests/frontend/integration/changeSettings.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+// #############################################################################
+// Change Settings behaviour
+
+var globals = require('./settings/globals');
+var cms = require('./helpers/cms')();
+var casperjs = require('casper');
+var xPath = casperjs.selectXPath;
```
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: unused variable (check output of `gulp lint`)

File: cms/tests/frontend/integration/changeSettings.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+                })
+        })
+
+        .waitForResource(/admin/)
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                //checks h1 tag if the language changed
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/changeSettings.js]: space after `//`

File: cms/tests/frontend/integration/changeSettings.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+                    test.assertEquals(this.fetchText('#content h1'), 'Benutzer-Einstellung ändern', 'German language applied');
+                })
+
+                .waitForSelector('#content-main', function () {
+
+                    test.assertExists('#id_language', 'Menu correct reloaded');
+
+                    this.click('#id_language');
+
+                    //changes back to english
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/changeSettings.js]: space missing again

File: cms/tests/frontend/integration/changeSettings.js [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+                })
+        })
+
+        .waitForResource(/admin/)
+
+        // have to wait for the sideframe again, cause the page reloads itself after saving the the language
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                 //checks h1 tag if the language changed
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/changeSettings.js]: space

File: cms/tests/frontend/integration/changeSettings.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        })
+        // click on User Settings
+        .waitUntilVisible('.cms-toolbar-item-navigation-hover a', function () {
+            this.click('.cms-toolbar-item-navigation-hover a[href$="/admin/cms/usersettings/"]');
+        })
+        // waits till Sideframe is open
+        .waitUntilVisible('.cms-sideframe-frame')
+        .withFrame(0, function () {
+            casper
+                .waitForSelector('#content-main', function () {
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/changeSettings.js]: this should be on the same line or @vxsx ?

**Comment by vxsx** [comment]
so far LGTM, wait for travis

**Comment by FinalAngel** [comment]
LGTM

File: cms/tests/frontend/integration/changeSettings.js (hunk 14 above)
Comment (vxsx) [inline cms/tests/frontend/integration/changeSettings.js]: i'm ok with this for now, this style is used often in the integration tests
we can clean it up in one sweep

**Comment by vxsx** [comment]
@genti94 
```
cms/tests/frontend/integration/changeSettings.js
  line 45  col 19  Missing semicolon.
  line 68  col 19  Missing semicolon.
  line 81  col 19  Missing semicolon.
  ⚠  3 warnings
```

**Pull request merged**